embark-bibtex.el

```diff
@@ -0,0 +1,140 @@
+;;
+;; Usage:
+;;
+;;    Do M-x embark-bibtex and start typing a search query when prompted.
+
+;;; Code:
+
+(require 'embark)
+(require 'bibtex-completion)
+
+(defcustom embark-bibtex-default-action 'embark-bibtex-open-any
+  "The default action for the `embark-bibtex` command."
+  :group 'bibtex-completion
+  :type 'function)
+
+(defmacro embark-bibtex-embarkify-action (action name)
+  "Wraps the function ACTION in another function named NAME which
+extracts the key from the candidate selected in embark and
+passes it to ACTION."
+  `(defun ,name (cand)
+     (interactive "s ")
+     (,action (list cand))))
+
+(embark-bibtex-embarkify-action bibtex-completion-open-any embark-bibtex-open-any)
+(embark-bibtex-embarkify-action bibtex-completion-open-pdf embark-bibtex-open-pdf)
+(embark-bibtex-embarkify-action bibtex-completion-open-url-or-doi embark-bibtex-open-url-or-doi)
+(embark-bibtex-embarkify-action bibtex-completion-insert-citation embark-bibtex-insert-citation)
+(embark-bibtex-embarkify-action bibtex-completion-insert-reference embark-bibtex-insert-reference)
+(embark-bibtex-embarkify-action bibtex-completion-insert-key embark-bibtex-insert-key)
+(embark-bibtex-embarkify-action bibtex-completion-insert-bibtex embark-bibtex-insert-bibtex)
+(embark-bibtex-embarkify-action bibtex-completion-add-PDF-attachment embark-bibtex-add-PDF-attachment)
+(embark-bibtex-embarkify-action bibtex-completion-edit-notes embark-bibtex-edit-notes)
+(embark-bibtex-embarkify-action bibtex-completion-show-entry embark-bibtex-show-entry)
+(embark-bibtex-embarkify-action bibtex-completion-add-pdf-to-library embark-bibtex-add-pdf-to-library)
+
+(defun embark-bibtex--get-candidates ()
+  "Return all keys from bibtex-completion-candidates."
+  (mapcar
+   (lambda (cand)
+     (cons (bibtex-completion-format-entry cand (1- (frame-width)))
+           (cdr (assoc "=key=" cand))))
+   (bibtex-completion-candidates)))
+
+;; TODO Broken, use (cdr (assoc (completing-read "bib" candidates) candidates)) form
+;;;###autoload
+(defun embark-bibtex (bib-entry)
+  "Search BibTeX entries using `completing-read' and embark
+actions."
+  (interactive
+   (list
+    (completing-read
+     "BibTeX entries: "
+     (lambda (string predicate action)
+       (if (eq action 'metadata)
+           '(metadata (category . bibtex))
+         (complete-with-action action (embark-bibtex--get-candidates) string predicate))))))
+  (apply embark-bibtex-default-action (list bib-entry)))
+
+(embark-define-keymap embark-bibtex-map
+  "Keymap for actions for bibtex."
+  ("p" embark-bibtex-open-pdf)
+  ("u" embark-bibtex-open-url-or-doi)
+  ("c" embark-bibtex-insert-citation)
+  ("r" embark-bibtex-insert-reference)
+  ("k" embark-bibtex-insert-key)
+  ("b" embark-bibtex-insert-bibtex)
+  ("a" embark-bibtex-add-PDF-attachment)
+  ("e" embark-bibtex-edit-notes)
+  ("s" embark-bibtex-show-entry)
+  ("l" embark-bibtex-add-pdf-to-library))
+
+(add-to-list 'embark-keymap-alist '(bib . embark-bibtex-map))
```

Comment:
4. At this point, this line is the *only dependency* with Embark. Remove it.  Then you are completely independent of Embark. This line can be written in the user config in order to make the keymap available to embark.

Reply:
So in the end, the idea for a user who has selectrum setup as their completion framework, they just install this package (or, ideally at some point, just bibtex-completion), and install and configure embark to use this keybinding, as so associated the bibtex-completion commands with the selected candidates?